### PR TITLE
Extended dmat

### PR DESCRIPTION
**Pull request by vserret**

### CODE REVIEW

**Comment by londumas** [comment]
@vserret , could you add a dot to float in the parser ?
And also the dimension ? i.e. [h^-1 Mpc]

**Comment by ngbusca** [comment]
@vserret did you voluntarily closed this pull request?

**Comment by vserret** [comment]
I am sorry. I reopen it

**Comment by londumas** [comment]
can you change "max rt" for "max rt [h^-1 Mpc]". ?

File: py/picca/cf.py [hunk 1]
```diff
@@ -112,7 +112,8 @@ def fast_cf(z1,r1,w1,d1,z2,r2,w2,d2,ang,same_half_plate):
     bt = (rt/rt_max*nt).astype(int)
     bins = bt + nt*bp
     if same_half_plate:
-        w = bp == 0
+        if x_correlation : w = bp == np/2
```
Comment (ngbusca) [inline py/picca/cf.py]: @vserret this assumes that rp_min = -rp_max...

File: py/picca/cf.py [hunk 2]
```diff
@@ -112,7 +112,8 @@ def fast_cf(z1,r1,w1,d1,z2,r2,w2,d2,ang,same_half_plate):
...
+        else : w = bp == 0
```
Comment (ngbusca) [inline py/picca/cf.py]: this assumes that rp_min = 0 (there's no guarantee that a user won't set rp_min for an auto-correlation)

File: py/picca/cf.py [hunk 3]
```diff
@@ -189,7 +191,8 @@ def fill_dmat(l1,l2,r1,r2,w1,w2,ang,wdm,dm,same_half_plate,order1,order2):
     we = w1[:,None]*w2
     we = we[w]
     if same_half_plate:
-        wsame = bp[w]==0
+	if x_correlation : wsame = bp[w]==np/2
+	else : wsame = bp[w]==0
```
Comment (ngbusca) [inline py/picca/cf.py]: same comment in these two lines

**Pull request closed without merging**